### PR TITLE
samples: Rename at_client -> at_host

### PR DESCRIPTION
**Pull request by bjda**
Renames the at_client sample to at_host, to match the name of the
library being demonstrated.

### CODE REVIEW

File: samples/nrf9160/at_host/README.rst [hunk 1]
```diff
@@ -1,19 +1,19 @@
-.. _at_client_sample:
+.. _at_host_sample:
 
-nRF9160: AT Client
-##################
+nRF9160: AT Host sample
+#######################
```
Comment (umapraseeda) [inline samples/nrf9160/at_host/README.rst:4]:
```suggestion
nRF9160: AT Host
################
```

**Review by rlubos — COMMENTED** [review]
Looks good, but perhaps mention in the `release-notes-latest.rst` that the sample was renamed?

**Review by lemrey — APPROVED** [review]
Did not check if anything was missing, but the diff looks good.
nit: I would use the `samples: nrf9160:` prefix to the commit title.

**Comment by martiles** [comment]
https://infocenter.nordicsemi.com/topic/ug_nrf91_dk_gsg/UG/nrf91_DK_gsg/testing_at_client.html
FYI. Documentation pages that reference at_client needs to be handled as well because of this change.

**Comment by umapraseeda** [comment]
> https://infocenter.nordicsemi.com/topic/ug_nrf91_dk_gsg/UG/nrf91_DK_gsg/testing_at_client.html
> https://infocenter.nordicsemi.com/pdf/nwp_034.pdf
> FYI. Documentation pages that reference at_client needs to be handled if this change is going to happen. (at_client is referenced in a lot of places)
Yes. Created a JIRA for that.

File: doc/nrf/releases/release-notes-latest.rst [hunk 2]
```diff
@@ -93,6 +93,9 @@ nRF9160
 
     * Added support for the nRF9160 pulse-density modulation (PDM) and inter-IC sound (I2S) peripherals in non-secure applications.
 
+  * :ref:`at_host_sample` sample:
+    * Renamed nRF9160: AT Client sample to :ref:`at_host_sample`.
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-latest.rst:97]: Suggesting a newline here to fix the output:
```suggestion
  * :ref:`at_host_sample` sample:

    * Renamed nRF9160: AT Client sample to :ref:`at_host_sample`.
```

File: samples/nrf9160/aws_fota/README.rst [hunk 3]
```diff
@@ -96,7 +96,7 @@ Add the certificates to the sample code:
 Use LTE Link Monitor to write the certificates to the kit:
    The nRF Connect `LTE Link Monitor`_ provides a certificate manager that you can use to store the certificates on your kit:
 
-   1. Make sure that you have the AT client sample programmed on your kit.
+   1. Make sure that you have the :ref:`at_host_sample` programmed on your kit.
```
Comment (umapraseeda) [inline samples/nrf9160/aws_fota/README.rst:99]:
```suggestion
   1. Make sure that you have the :ref:`at_host_sample` sample programmed on your kit.
```

**Review by umapraseeda — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**